### PR TITLE
Update SQL plugin relevance functions documentation

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Signed-off-by: Yury Fridlyand <yuryf@bitquilltech.com>

## Changes include
* Add new Relevance Based search functions
* Move `PPL` section under `SQL Plugin`
* Reorganize `SQL Plugin` and remove duplicates
* Update/fix/remove obsolete and outdated info
* Update links

## TBD
- [ ] Add (copy from repo) or don't add more info about JDBC driver
- [ ] Confirm moving `PPL` to `SQL Plugin` or move it back or ...

## Note
https://github.com/opensearch-project/sql/pull/665 should be merged first to have all valid links.

### CODE REVIEW

**Comment by Yury-Fridlyand** [comment]
[Full-Text Search - OpenSearch documentation.pdf](https://github.com/Yury-Fridlyand/documentation-website/files/9059181/Full-Text.Search.-.OpenSearch.documentation.pdf)

**Comment by Yury-Fridlyand** [comment]
TODO:
* mention PPL in SQL
* Validate and/or delete information about [`delete`](https://opensearch.org/docs/latest/search-plugins/sql/delete/)
* Add more functions and examples
* Mention `explain` request

**Review by MitchellGale-BitQuill — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]
question: why move PPL section under SQL plugin? i believe they were in different sections because their use cases are different

**Comment by Yury-Fridlyand** [comment]
> question: why move PPL section under SQL plugin? i believe they were in different sections because their use cases are different

Well, yes. Following this approach, I have to return PPL back.
The idea was: PPL provided and maintained by the SQL Plugin.

I'm open to move it back to Observability Plugin or add clearer references there or whatever we can find good for a user.

File: _search-plugins/sql/sql/jdbc.md [hunk 1]
```diff
@@ -10,3 +11,7 @@ nav_order: 71
 The Java Database Connectivity (JDBC) driver lets you integrate OpenSearch with your favorite business intelligence (BI) applications.
 
 For information on downloading and using the JAR file, see [the SQL repository on GitHub](https://github.com/opensearch-project/sql/tree/master/sql-jdbc).
+
+## Connecing to Tableau
```
Comment (forestmvey) [inline _search-plugins/sql/sql/jdbc.md:15]: Connecting*

File: _observability-plugin/index.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ Observability is collection of plugins and applications that let you visualize d
 
 Your experience of exploring data might differ, but if you're new to exploring data to create visualizations, we recommend trying a workflow like the following:
 
-1. Explore data over a certain timeframe using [Piped Processing Language]({{site.url}}{{site.baseurl}}/observability-plugin/ppl/index).
+1. Explore data over a certain timeframe using [Piped Processing Language]({{site.url}}{{site.baseurl}}/search-plugins/sql/ppl/index).
```
Comment (acarbonetto) [inline _observability-plugin/index.md:19]: We should keep a page in the Observability Plugin section that points to the SQL Plugin/PPL section

File: _search-plugins/sql/cli.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
 ---
 layout: default
 title: SQL CLI
-parent: SQL
-nav_order: 2
+parent: SQL Plugin - SQL & PPL
```
Comment (acarbonetto) [inline _search-plugins/sql/cli.md:4]: How about this for ordering of the SQL Plug - SQL & PPL section:
```
SQL PLugin - SQL & PPL
  Endpoint
  Protocol
  SQL CLI (Rename to CLI or "Command-Line Interface")
  SQL
    Query Workbench
    Basic Queries
    Complex Queries
    JSON Support
    Metadata Queries
    Aggregation Functions
    Delete
    JDBC Driver
    ODBC Driver
  PPL - Piped Processing Language
    Functions (Split into two pages: "General Syntax" and "PPL Functions")
  Identifiers
  Data Types
  Functions
  Full-Text Search
  Settings
  Troubleshooting
  Monitoring
  Limitations
  
```

**Review by acarbonetto — CHANGES_REQUESTED** [review]

**Comment by acarbonetto** [comment]
> > question: why move PPL section under SQL plugin? i believe they were in different sections because their use cases are different
> 
> Well, yes. Following this approach, I have to return PPL back. The idea was: PPL provided and maintained by the SQL Plugin.
> 
> I'm open to move it back to Observability Plugin or add clearer references there or whatever we can find good for a user.

It makes sense to move it to the SQL Plugin section, as the PPL syntax is mostly just a variation of the SQL query syntax, but also shares a lot of the same tooling. 

@Yury-Fridlyand let's make sure to preserve a section in the Observability plugin that refers tot the PPL section of the SQL plugin, so that we don't lose that reference.  Otherwise, I think most content should be in the SQL plugin section to avoid duplication.

**Comment by forestmvey** [comment]
Not your code but this shouldn't be in the table of contents pane:

[sql/ppl/functions.md#L899](https://github.com/Yury-Fridlyand/documentation-website/blob/55f4652fc784d749ccfec52f84601910762e6d45/_search-plugins/sql/ppl/functions.md?plain=1#L899)

**Review by forestmvey — CHANGES_REQUESTED** [review]
Few minor changes requested.

File: _search-plugins/sql/cli.md [hunk 4]
```diff
@@ -33,25 +33,15 @@ SQL CLI has the following features:
 
 Launch your local OpenSearch instance and make sure you have the SQL plugin installed.
 
-To install the SQL CLI:
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply _search-plugins/sql/cli.md:36]: These steps were required to build/launch CLI locally before it published on pip/pypi.

**Review by forestmvey — APPROVED** [review]

**Comment by Yury-Fridlyand** [comment]
Please, see most recent changes made in c4a42dd and 60fd14b.

File: _search-plugins/sql/settings.md [hunk 5]
```diff
@@ -20,6 +20,22 @@ PUT _cluster/settings
 }
 ```
 
+`PPL` and `SQL` could be deactiaved independently.
+There is another format of request which could be used:
```
Comment (MaxKsyunz) [inline _search-plugins/sql/settings.md:24]: Is this line necessary? I'm not sure what it's trying to say.

File: _search-plugins/sql/limitation.md [hunk 6]
```diff
@@ -1,62 +1,18 @@
...
-nav_order: 18
+parent: SQL Plugin - SQL & PPL
+nav_order: 99
 ---
 
 # Limitations
 
 The SQL plugin has the following limitations:
 
-## SELECT FROM WHERE
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply _search-plugins/sql/limitation.md:12]: No, V2 engine can do `SELECT 1;`.
Good point, I'll recover this in the end with corresponding note.

File: _search-plugins/sql/limitation.md [hunk 7]
```diff
@@ -1,62 +1,18 @@
...
-
-The `WHERE` clause does not support expressions. For example, `SELECT FlightNum FROM opensearch_dashboards_sample_data_flights where (AvgTicketPrice + 100) <= 1000` is not supported.
-
-
-### Aggregation over expression is not supported
+## Aggregation over expression is not supported
 
 You can only apply aggregation on fields, aggregations can't accept an expression as a parameter. For example, `avg(log(age))` is not supported.
 
-
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply _search-plugins/sql/limitation.md:28]: No. I'm not sure that there is a valid response, but at least there is no error/failures:
```json
$ curl -H 'Content-Type: application/x-ndjson' -XPOST 'http://localhost:9200/conflict_index_1/_doc/?pretty' -d '{ "a": { "b": 1 } }'
{
  "_index" : "conflict_index_1",
  "_id" : "OwHfI4IBUwd-1Kul7tnr",
  "_version" : 1,
  "result" : "created",
  "_shards" : {
    "total" : 2,
    "successful" : 1,
    "failed" : 0
  },
  "_seq_no" : 0,
  "_primary_term" : 1
}

$ curl -H 'Content-Type: application/x-ndjson' -XPOST 'http://localhost:9200/conflict_index_2/_doc/?pretty' -d '{ "a": { "b": 1, "c2": 2 } }'
{
  "_index" : "conflict_index_2",
  "_id" : "PAHgI4IBUwd-1KulH9mg",
  "_version" : 1,
  "result" : "created",
  "_shards" : {
    "total" : 2,
    "successful" : 1,
    "failed" : 0
  },
  "_seq_no" : 0,
  "_primary_term" : 1
}
```
```console
opensearchsql> SELECT * FROM conflict_index*;
fetched rows / total rows = 2/2
+-------------------+
| a                 |
|-------------------|
| {'b': 1}          |
| {'b': 1, 'c2': 2} |
+-------------------+
```

**Review by acarbonetto — CHANGES_REQUESTED** [review]

File: _search-plugins/sql/endpoints.md [hunk 8]
```diff
@@ -66,45 +50,17 @@ Explain query:
...
-    "excludes": []
-  }
-}
+```console
+>> curl -H 'Content-Type: application/json' -X POST localhost:9200/_plugins/_ppl/_explain -d '{
+  "query" : "source=accounts | fields firstname, lastname"
+}'
 ```
 
+V2 query engine not only runs queries in OpenSearch engine but also supports post-processing for complicated queries. Accordingly, explain output is no longer pure OpenSearch DSL, but also includes query plan information from the V2 query engine.
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply _search-plugins/sql/endpoints.md:61]: Thanks, I move this phrase to `limitations`

**Pull request merged**